### ISSUE TITLE
How did you visualize the Superpixel MNIST?

### ISSUE THREAD

**Issue by Felijasha** [rusty1s/pytorch_geometric#320]
## ❓ Questions & Help

Hey :)
I read your paper SplineCNN: Fast Geometric Deep Learning with Continuous B-Spline Kernels.
I saw a visualization of the Superpixel MNIST. Do you have any idea how I can visualize a generated Superpixel example?

Thank you :)

**Comment by rusty1s** [comment]
I have found the script [here](https://github.com/rusty1s/pytorch_geometric/blob/da78a9bc6074913713e0fffd908a4365dfe20ef9/torch_geometric/visualization/mnist_graph.py), but it may be a bit hacky and hard to read.

**Comment by Felijasha** [comment]
Thank you for your quick answer :) I will have alook at it!

**Comment by phaniram-sayapaneni** [comment]
the script doesn't work

**Comment by rusty1s** [comment]
Yes, it is quite old and may not work with newer PyG versions. Sorry for the inconveniences!

**Comment by duyguislakoglu** [comment]
If you set dummy y (in this case 0) to the train dataset, you can visualize the Superpixel MNIST as follows:
(Modifying https://discuss.pytorch.org/t/pytorch-geometric/44994/6) 

![Ekran Resmi 2020-04-29 22 26 13](https://user-images.githubusercontent.com/37613999/80639361-9178c900-8a6a-11ea-8ef9-875c23e6acaa.png)

import networkx as nx
import matplotlib.pyplot as plt 
from torch_geometric.utils.convert import to_networkx

train_dataset = MNISTSuperpixels("drive/My Drive/mnist", True, transform=transform)
for i in range(10,16):
  mnist = to_networkx(train_dataset[i])
  train_dataset.y = torch.zeros(75)
  node_labels = train_dataset.y.numpy()
 
  plt.figure(1,figsize=(5,5)) 
  nx.draw(mnist, cmap=plt.get_cmap('Set2'),node_color = node_labels,node_size=30,   linewidths=6)
  plt.show()

**Comment by rusty1s** [comment]
You can also make use of the `pos` attribute of nodes for better positions of nodes.

**Comment by Somedaywilldo** [comment]
Hey guys!

I believe this is what everybody wants while trying to visualize the mnist_superpixel dataset.

![image](https://user-images.githubusercontent.com/22238123/116396769-56b19880-a858-11eb-9a07-9ef64a901de3.png)

Actually it took me quite some time to make it looks perfect since I found the default axis direction of networkx.draw() is wried, the y-axis of data.pos needs to be flipped over, also the index of networkx graph is not in order, so it requires some re-indexing operations.

And I think my solution is very easy to read and not hacky at all : ), so I'd love to share it with everybody, created a pull request here #2488, please enjoy!
